src/utils/createTabNavigator.js

```diff
@@ -70,6 +72,30 @@ export default function createTabNavigator(TabView: React.ComponentType<*>) {
       return route.routeName;
     };
 
+    _getAccessibilityLabelText = route => {
+      const { descriptors } = this.props;
+      const descriptor = descriptors[route.key];
+      const options = descriptor.options;
+
+      if (options.tabBarAccessibilityLabel) {
+        return options.tabBarAccessibilityLabel;
+      }
+
+      return this._getLabelText({ route });
+    };
```

Comment:
Can you change this to following so we make sure to reuse label only if it's a string:

```js
const label = this._getLabelText({ route });

if (typeof label === 'string') {
  return label;
}
```